### PR TITLE
Fix mobile app update issue

### PR DESCRIPTION
**Pull request by dasuni-30**
### Related Issues
- https://github.com/wso2-enterprise/wso2-iam-internal/issues/416

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by chaminjay — APPROVED** [review]

File: apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx [hunk 1]
```diff
@@ -937,7 +937,7 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
                 backChannelLogoutUrl: values.get("backChannelLogoutUrl"),
                 frontChannelLogoutUrl: values.get("frontChannelLogoutUrl")
             },
-            publicClient: values.get("supportPublicClients")?.length > 0,
+            publicClient: !isMobileApplication ? values.get("supportPublicClients")?.length > 0 : true,
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx:940]:
```suggestion
            publicClient: isMobileApplication || values.get("supportPublicClients")?.length > 0,
```

Comment (brionmario) [reply apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx:940]: +1 or use `Nullish coalescing operator (??)` to simplify the ternary.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx (hunk 1 above)
Comment (dasuni-30) [reply apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx:940]: Both methods are correct.

Comment (pavinduLakshan) [reply apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx:940]: Both ways are functionally correct. The suggestion was to improve readability :)

**Pull request merged**